imgui-SFML.cpp

```diff
@@ -680,21 +666,18 @@ void Update(const sf::Vector2i& mousePos, const sf::Vector2f& displaySize, sf::T
     io.DisplaySize = ImVec2(displaySize.x, displaySize.y);
     io.DeltaTime = dt.asSeconds();
 
-    if (s_currWindowCtx->windowHasFocus) {
-        if (io.WantSetMousePos) {
-            sf::Vector2i newMousePos(static_cast<int>(io.MousePos.x),
-                                     static_cast<int>(io.MousePos.y));
-            sf::Mouse::setPosition(newMousePos);
-        } else {
-            io.MousePos = ImVec2(static_cast<float>(mousePos.x), static_cast<float>(mousePos.y));
-        }
-        for (unsigned int i = 0; i < 3; i++) {
-            io.MouseDown[i] = s_currWindowCtx->touchDown[i] || sf::Touch::isDown(i) ||
-                              s_currWindowCtx->mousePressed[i] ||
-                              sf::Mouse::isButtonPressed((sf::Mouse::Button)i);
-            s_currWindowCtx->mousePressed[i] = false;
-            s_currWindowCtx->touchDown[i] = false;
-        }
+    if (io.WantSetMousePos) {
+        sf::Vector2i newMousePos(static_cast<int>(io.MousePos.x), static_cast<int>(io.MousePos.y));
+        sf::Mouse::setPosition(newMousePos);
+    } else {
+        io.MousePos = ImVec2(static_cast<float>(mousePos.x), static_cast<float>(mousePos.y));
+    }
+    for (unsigned int i = 0; i < 3; i++) {
+        io.MouseDown[i] = s_currWindowCtx->touchDown[i] || sf::Touch::isDown(i) ||
+                          s_currWindowCtx->mousePressed[i] ||
+                          sf::Mouse::isButtonPressed((sf::Mouse::Button)i);
```

Comment:
@oprypin  I see what you're saying now - it's because this directly queries the mouse state here which bypasses the event queue.

Ideally think this should use events instead?

Reply:
Yeah, it looks like this is what causes problems, thanks.
I guess I should really rework how ProcessEvent/Update/Render etc. work and not use any direct queries since they don't respect window focus. 

Of course, maybe if I check `s_currWindowCtx.window->hasFocus()` it will solve the problem, but it feels somewhat hacky.